include/display/graphics.h

```diff
@@ -0,0 +1,410 @@
+ * Exposed mostly for writing libraries and custom functions.
+ * 
+ * \param txt_fmt Text format num that correspondes with a higher level enum
+ * \param x,The y coordinates of the top left corner of the string
+ * \param y The y coordinates of the top left corner of the string
+ * \param fmt  Format string
+ * \param args List of arguments for the format string
+ */
+void display_vprintf_at(uint8_t txt_fmt, const int16_t x, const int16_t y, const char* fmt, va_list args);
+
+/**
+ * Print a formatted string to the screen on the specified line
+ *
+ * Same as `display_printf` except that this uses a `va_list` instead of the
+ * ellipsis operator so this can be used by other functions.
+ * 
+ * Will default to a medium sized font by default if invalid text_format is given.
+ * Exposed mostly for writing libraries and custom functions.
+ * 
+ * \param txt_fmt Text format num that correspondes with a higher level enum
+ * \param line The line number on which to print
+ * \param fmt  Format string
+ * \param args List of arguments for the format string
+ */
+void display_vprintf(uint8_t txt_fmt, const int16_t line, const char* fmt, va_list args);
+
+/**
+ * Print a normal unformatted string to the screen at the specified coordinates
+ * 
+ * Will default to a medium sized font by default if invalid text_format is given.
+ * 
+ * Text formats medium_center and large_center will default to medium and large respectively.
+ * 
+ * \param txt_fmt Text format enum that determines if the text is small, medium, and large.
+ * \param x,The y coordinates of the top left corner of the string
+ * \param y The y coordinates of the top left corner of the string
+ * \param opacity The opacity of the text to display
+ * \param format The text to display, with formatting parameters afterwards
+ * \param args List of arguments for the format string
+ */
+void display_opacity_vprintf( int32_t xpos, int32_t ypos, uint32_t opacity, const char *format, va_list args);
+
+typedef enum touch_event_e { 
+	E_TOUCH_EVENT_RELEASE, 
+	E_TOUCH_EVENT_PRESS, 
+	E_TOUCH_EVENT_PRESS_AND_HOLD 
+} touch_event_e_t;
+
+typedef void (*touch_event_cb_fn_t)(int16_t, int16_t);
+
+typedef struct touch_event_position_data_s {
+	int16_t x;
+	int16_t y;
+} touch_event_position_data_s_t;
+
+/**
+ * Gets the x coordinate of the last touch.
+ * 
+ * \return The x coordinate of the last touch.
+ */
+int16_t touch_last_x(void);
+
+/**
+ * Gets the y coordinate of the last touch.
+ * 
+ * \return The y coordinate of the last touch.
+ */
+int16_t touch_last_y(void);
+
+/**
+ * Gets the touch status of the last touch of the screen.
+ * 
+ * \return The touch_event_e_t enum specifier that indicates the last touch status of the screen (E_TOUCH_EVENT_RELEASE, E_TOUCH_EVENT_PRESS, or E_TOUCH_EVENT_PRESS_AND_HOLD). 
+ */
+touch_event_e_t touch_last_event(void);
+
+/**
+ * Initializes the touch callback linked lists for different touch statuses.
+ */
+void _set_up_touch_callback_storage();
```

Comment:
Is this an implementation detail? If so it should be static and not be in the header.